iso.sh

```diff
@@ -0,0 +1,159 @@
+    mv "$isofs/boot/vmlinuz-"* "$isofs/boot/vmlinuz"
+
+    # Make some symbolink links because we don't have packages for these binaries (yet)
+    ln -s /bin/busybox "$isofs/bin/hostname"
+    ln -s /bin/busybox "$isofs/bin/vi"
+
+    # Copy the overlay directory into the ISO, overwriting existing files
+    cp -r "$overlay"/* "$isofs"/
+
+    # Cleanup the unneeded files
+    rm -rf \
+        "$isofs"/boot/System.map-* \
+        "$isofs"/boot/config-* \
+        "$isofs"/var/nest
+
+    # Move it to the Linux mountpoint
+    mv "$isofs"/* "$mnt_linux/"
+}
+
+# Generates the Grub configuration (grub.cfg)
+#
+# Takes the name of the root partition as parameter
+function generate_grub_cfg() {
+    cat << EOF
+menuentry 'Live' {
+	linux /boot/vmlinuz rootwait root=$1 quiet
+}
+
+menuentry 'Graphical install' {
+	linux /boot/vmlinuz rootwait root=$1 quiet
+}
+
+menuentry 'Manual install' {
+	linux /boot/vmlinuz rootwait root=$1 quiet
+}
+EOF
+}
+
+# Creates a bootable ISO and all the needed files to make it (like grub config)
+function make_iso() {
+    dd if=/dev/zero of="$build_iso" count=300 bs=1048576
+
+    # Create two partitions:
+    #  * EFI system partition (EF00), 50MiB in size
+    #  * Linux partition, filling the remaining space
+    parted --script "$build_iso" \
+        mklabel msdos \
+        mkpart p fat32 1 50MiB \
+        mkpart p ext2 50MiB 100% \
+        set 1 esp on \
+        set 2 boot on
+
+    sync
+
+    losetup -fP "$build_iso"
+
+    declare device_list=$(losetup -ln)
+    declare device="$(echo $device_list | grep "$build_iso" | awk '{print $1}' | head -n 1)"
+    declare partition_efi=${device}p1
+    declare partition_linux=${device}p2
+
+    echo "Device: $device"
+    echo "Partition EFI: $partition_efi"
+    echo "Partition Linux: $partition_linux"
+
+    # Create and mount the partitions
+    mkfs.fat -F32 "$partition_efi" && sync
+    mkfs.ext2 "$partition_linux" && sync
+
+    mount -t vfat "$partition_efi" "$mnt_efi"
+    mount -t ext2 "$partition_linux" "$mnt_linux"
+
+    # Populate the content of the partition
+    populate_isofs
+
+    # Install EFI grub
+    grub-install --target=x86_64-efi --themes= --recheck --removable --efi-directory="$mnt_efi" --boot-directory="$mnt_linux/boot" && sync
+
+    # Install BIOS grub
+    grub-install --target=i386-pc --themes= --recheck --boot-directory="$mnt_linux/boot" "$device" && sync
+
+    # Retrieve main partition's UUID
+    partuuid=$(blkid -s PARTUUID -o value "$partition_linux")
+
+    echo "PartUUID=$partuuid"
+
+    # Generate grub partition
+    #
+    # We use a partition UUID here so it's unambiguous which partition is the root
+    # one.
+    generate_grub_cfg "PARTUUID=$partuuid" >> "$mnt_linux/boot/grub/grub.cfg"
+
+    # Unmount and remove the loopback devices
+    umount -R "$mnt_linux"
+    umount -R "$mnt_efi"
+    losetup -d "$device"
+
+    sync
+
+    # Move the iso to it's final destination
```

Comment:
Should be `its`, not `it's`

Reply:
I make this error all the time :( Thanks!